### PR TITLE
TYP: changed variable cat_array to cat_array_list in dtypes.py and arr_list in categorical.py

### PR DESCRIPTION
**Pull request by nickleus27**
xref #37715

My apologies, I pushed hastily on my previous branch assign_err and did not check the diff of the files i was pushing. I accidentally pulled when i should have merged and I pushed a bunch of other peoples commits.  These are my previous changes.

### CODE REVIEW

**Comment by jreback** [comment]
umm what are these in reference to?

**Comment by nickleus27** [comment]
@jreback #44373

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -465,9 +465,11 @@ def _hash_categories(self) -> int:
                 [cat_array, np.arange(len(cat_array), dtype=cat_array.dtype)]
             )
         else:
-            # error: Incompatible types in assignment (expression has type
-            # "List[ndarray]", variable has type "ndarray")
-            cat_array = [cat_array]  # type: ignore[assignment]
+            cat_array_list = [cat_array]
+            combined_hashed = combine_hash_arrays(
```
Comment (jreback) [inline pandas/core/dtypes/dtypes.py:469]: umm can you avoid repeating the code here on L 469

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @nickleus27

**Pull request merged**